src/utils.rs

```diff
@@ -90,6 +92,22 @@ pub async fn gc_partitions(context: &DefaultSeafowlContext) {
     }
 }
 
+// Group adjacent partitions using the provided function
+pub fn group_partitions<F>(
+    partitions: Vec<SeafowlPartition>,
+    f: F,
+) -> Vec<(bool, Vec<SeafowlPartition>)>
+where
+    F: FnMut(&SeafowlPartition) -> bool,
```

Comment:
Can this be `Fn` instead of `FnMut`, given that it doesn't mutate `partitions_to_filter`/`partitions_to_update`? Not sure if this will result in any compiler optimizations though.

Reply:
Yup, changing.